jetty-core/jetty-io/src/test/java/org/eclipse/jetty/io/IOResourcesTest.java

```diff
@@ -50,6 +51,8 @@ public void tearDown()
         assertThat("Leaks: " + bufferPool.dumpLeaks(), bufferPool.getLeaks().size(), is(0));
     }
 
+    // This Resource impl has getPath() and newInputStream() return null so the only way for IOResources
+    // to read its contents is to call newContentSource().
```

Comment:
We'd still need a case to test where getPath() returns some path.

In that case, since this Resource implements `Content.Source.Factory`, it is still expected that `newContentSource()` is called.

This is very important so we can rule out wrong data being sent at some point:

Imagine someone who uses Path objects now in this class to reference to some virtual file system.

I know you could  always blame them for not following the spec, but it is what it is: Eventually, someone re-arranges the checks in Jetty's `IOResources`, and because `Path` is non-null, we suddenly go down a completely different route, and treat `Path` as a path to the underlying file system.

Reply:
Right, so I've made `getPath()` and `newInputStream()` throw to make sure only `newContentSource()` can be used to read that resource.